src/integrations/TiktokAds/util.js

```diff
@@ -0,0 +1,68 @@
+import { constructPayload } from '../../utils/utils';
+import { PARTNER_NAME, trackMapping } from './constants';
+import { removeUndefinedAndNullValues } from '../../utils/commonUtils';
+
+const getContents = (message) => {
+    const contents = [];
+    const { properties } = message;
+    // eslint-disable-next-line camelcase
+    const { products, content_type, contentType } = properties;
+    if (products && Array.isArray(products) && products.length > 0) {
```

Comment:
```suggestion
if (Array.isArray(products)) {
```
This should take care of it. And if at all `products` is an empty array, `.forEach` loop will not create a problem for us.

Reply:
Done